### ISSUE TITLE
b79250.sh failing with uncaught overflow on non-windows

### ISSUE THREAD

**Issue by richardlford** [dotnet/coreclr#2333]
The following test:

- JIT/Regression/CLR-x86-JIT/V1-M12-Beta2/b79250/b79250/b79250.sh

is failing on non-windows platforms with

```
    Unhandled Exception: System.OverflowException: Arithmetic operation resulted in an overflow.
```

**Comment by janvorli** [comment]
I will investigate it

**Comment by RussKeldorph** [comment]
@janvorli pls re-label if it's not a VM bug and you want to hand off.

**Comment by janvorli** [comment]
Fixed by #2543
